server/src/test/java/org/elasticsearch/action/admin/cluster/state/ClusterStateApiTests.java

```diff
@@ -52,10 +53,7 @@ public void testWaitForMetadataVersion() throws Exception {
         updateSettingsRequest.transientSettings(Settings.builder().put("cluster.max_shards_per_node", 999));
         assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
 
-        assertBusy(() -> {
-            assertThat(future2.isDone(), is(true));
-        });
-        ClusterStateResponse response = future2.actionGet();
+        response = future2.get(10L, TimeUnit.SECONDS);
```

Comment:
I kept this test cleanup even though it's not necessary now with the transport action changes reverted because busy asserting on futures was just weird ...